app/src/main/java/com/github/multimatum_team/multimatum/model/User.kt

```diff
@@ -2,10 +2,23 @@ package com.github.multimatum_team.multimatum.model
 
 typealias UserID = String
 
+/**
+ * Model for users.
+ * @param id an unique identifier provided by the user repository
+ */
 sealed class User(open val id: UserID)
```

Comment:
Should one really be able to instantiate `User`? Would it not be better to make it an interface?

Reply:
Sealed classes cannot be instantiated, also interfaces cannot define attributes

Reply:
Indeed it cannot be instantiated, my bad. But I still find it a bit weird to use a class for this purpose. And interfaces can define attributes, at least
```Kotlin
sealed interface Foo {
    val value: Int
}
data class Bar(override val value: Int): Foo
```
compiles.

Reply:
Ah, I didn't know Kotlin could do that. What would be the benefit of defining an interface rather than a class in this case?

Reply:
It would make code clearer (at least in my opinion). In the subclasses it is weird to override `id` and simultaneously pass it to the constructor of the supertype.